ui/app/components/fs/file.js

```diff
@@ -160,4 +165,34 @@ export default class File extends Component {
   failoverToServer() {
     this.set('useServer', true);
   }
+
+  @action
+  downloadFile() {
+    const timing = this.useServer ? this.serverTimeout : this.clientTimeout;
+    const fileDownload = url =>
+      RSVP.race([this.token.authorizedRequest(url), timeout(timing)])
+        .then(
+          response => {
+            if (!response || !response.ok) {
+              this.nextErrorState(response);
+            }
+            return response;
+          },
+          error => this.nextErrorState(error)
+        )
+        .then(response => response.blob())
+        .then(blob => {
+          var url = window.URL.createObjectURL(blob);
+          var a = document.createElement('a');
+          a.href = url;
+          a.target = '_blank';
+          a.rel = 'noopener noreferrer';
+          a.download = this.file;
+          document.body.appendChild(a); // we need to append the element to the dom -> otherwise it will not work in firefox
+          a.click();
+          a.remove(); //afterwards we remove the element again
+          window.URL.revokeObjectURL(url);
+        });
+    fileDownload(this.catUrlWithoutRegion);
```

Comment:
Why don't we want the region in the URL for this?

Reply:
Because inside this function I make call `this.token.authorizedRequest(url)` which will add token to headers and region to query (as it done in catUrl).

Reply:
Oh, right, it's been a minute since I wrote that code 🤦